### DISCUSSION TITLE
Budget Editor Help

### DISCUSSION THREAD

**Discussion by Macoland** [jerrymjones/MonthlyBudgetEditor#11] [General]
Hello there, I first wanted to thank you for writing this amazing code.  I am trying to move from Quicken to Money Dance and all three of your extensions are amazing.  I do have one question that maybe you can help me with.  With Quicken I liked using the budget tool at your bars extension replicates it well.  I am confused about how the expense category works though.  It looks like all expenses roll up to that category.  In my screenshot I can see that Amazon and Auto and Transport and all others roll up to expenses.  And the total budget value of $1500 is correct.  But when I go to the summary page with the bars you can see that I am 456.78 over budget for Amazon, but the expenses category only shows $60.82 as being spent.  I am not really sure why that would be happening.  Any help would be appreciated.  Thanks again!  

![2023-05-14_23-17-13](https://github.com/jerrymjones/MonthlyBudgetEditor/assets/9373267/c18148db-7845-4e71-80c7-91b809d88b08)

![2023-05-14_23-17-55](https://github.com/jerrymjones/MonthlyBudgetEditor/assets/9373267/5e967c68-83fe-48f6-bc83-28a59e5228db)

**Comment by jerrymjones** [comment]
Hi Macoland, you're correct - that doesn't look right. In developing the extensions I often saw stuff like this and it was almost always because I had transactions that weren't assigned to the proper category. If you hover over the bars with the mouse cursor additional detail will be displayed.  Does that information shed any light? You can also run the Moneydance _**Income and Expense, Detailed**_ report. Does that show anything useful? Is it as you expect? Let me know what you find, if it's a bug it needs to be fixed :-)

**Comment by Macoland** [comment]
Thanks for the response.  I am trying to piece this together and I think I found a couple of possible problems.  

1st, when I hover over the expenses category it shows a lot more than I was expecting.  What I was looking for was the bar that represented only the items that I had put a budget next to (if that exists).  For example, in the screenshots I have no budget value next to MISC (I do have them next to Walmart and Sam's Club), but it is showing up in the list.  

![00](https://github.com/jerrymjones/MonthlyBudgetEditor/assets/9373267/98644103-d44f-4c72-996c-b311a7466431)



![01](https://github.com/jerrymjones/MonthlyBudgetEditor/assets/9373267/48aa2add-d194-48b7-8f23-4693b7e4971a)



2nd, the MISC value seems to be messing up the rest of my numbers due to the negative value.  The reason that this value even exists is because when I was testing moving things over I needed to move over the starting account balance.  So I created a transaction with the date of 12/31/2021 but it is showing up in the list of expenses from "this month."  So I guess "this month" looks at something other than the date of the item and perhaps at some other data that shows exactly when it was put into the Moneydance?  

![02](https://github.com/jerrymjones/MonthlyBudgetEditor/assets/9373267/cec3aeca-c991-451a-9c9a-510b6b336308)


Thanks again for your help on this!!

**Comment by jerrymjones** [comment]
Thanks for the update. I'm not home right now but I will look into this some more this afternoon. From what you are reporting it does look like you may have uncovered a bug. The popup does show all contributions to the summary total whether budgeted or not (I may need to rethink that), negative numbers should be fine, but it shouldn't pick up an out of date expense for "this month".

**Comment by Macoland** [comment]
No worries at all.  For me ideally there would be an "All expenses" bar which had everything and a "Budgeted Expenses" bar that would only show the culmination of parent categories from the budget sheet that had a value.  I might not be a normal user, but some expenses I don't care about as much as others.  

By the way, is there a good way to donate to you?  I would love to send a little money your way for taking the time to develop these helpful extensions.

**Comment by jerrymjones** [comment]
Macoland, thanks for the offer to make a donation. At the current time I don't have a way to do that. I hope that others find these useful and that is sufficient payment for me. At the moment, you're having a problem anyway so I don't think I should be rewarded :-) 

Just looking at all this, there is another way to handle the initial balance. For bank accounts there is an Initial balance field and for credit cards there is an Initial debt field when you edit the accounts. You could take care of the $5506.42 with one of these but honestly, other than maybe for reporting I can't see that it matters. Either way ought to work. I've tried playing with similar entries and so far I haven't duplicated the problem you are seeing. For the Amazon category, can you send me a list of the transaction amounts for this month that were charged? Just the amounts, not the details of the transactions. Also, I'm assuming you are using the latest releases of the extensions, correct? What version of Moneydance are you on and on what OS?

**Comment by Macoland** [comment]
Thanks for the update on the initial balance, I edited the account and put the balance there and everything looks good.  So from that standpoint I don't think there is a bug, I think I just had my stuff in the wrong place.  

I think everything else is good.  You asked about the Amazon category and all of that category is correct based on what I have input.  I guess at this point my only request would be to add another bar that only calculates based on budget categories that have values.  I have it setup that way in Quicken so it would be a nice to have, but if that is too much effort I totally understand.  

In answer to your other questions, I am using the latest versions, I just built them last week in a Linux VM.  I am on Moneydance 2023.1 (the latest) and I am running it on Windows 11.  

Thanks again for your prompt responses and help.

**Comment by jerrymjones** [comment]
I'm glad that worked but I still plan to review the code a bit more as it seems like it should have still worked correctly with the negative Misc entry.  I'll look at your proposed changes, they may not be that difficult to add. It will probably take a few weeks to get this done and I may come back here and ask you a few more questions.

**Comment by jerrymjones** [comment]
Macoland, I took a quick look and I think it would be fairly easy to add a checkbox in the Edit dialog to only include budgeted items in the spending for the bars (ignoring any unbudgeted spending). This would apply to all bars selected for display. Would this work for you?

**Reply by Macoland** [reply]
Ok, so the behavior would be that if I check the box (using the screenshot from the first post) the Amazon bar would stay the same since it is a budgeted category, but the Expenses category would change because it currently includes everything and it would change to only be budgeted items.  If that is the case, I think that would be a perfect solution.

**Reply by jerrymjones** [reply]
I'll start working on this. Give me a couple of weeks as we're travelling and visiting family.

**Reply by Macoland** [reply]
No worries at all, enjoy time with family!

**Comment by jerrymjones** [comment]
Macoland,

Sorry for the delay in doing this. When you get a chance can you try this and see if it does what you want? You will need to unzip the file first then load the .mxt file. In the Edit dialog there is a new checkbox for "Exclude unbudgeted categories". Try checking that and seeing if the results are as desired. Once we're happy, I'l post it as a regular release.

[budgetbars.zip](https://github.com/jerrymjones/MonthlyBudgetEditor/files/11716052/budgetbars.zip)

**Comment by Macoland** [comment]
@jerrymjones I just tried the new mxt file and it works PERFECTLY!  Thank you so much for the effort you put into this.  You have made this software much more useful!  

Also, I hope you had a great time traveling and with your family!

**Reply by jerrymjones** [reply]
Fantastic! You're welcome. I'll "officially" post it in the next few days. Feel free to reach out again if you find any issues or have other suggestions.
